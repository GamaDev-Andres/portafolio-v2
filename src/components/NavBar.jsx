import { Link } from 'react-router-dom'
import ListOfLinks from './ListOfLinks'

const NavBar = () => {
  return (
    <nav className="flex justify-center sm:justify-between flex-col md:flex-row sm:fixed sm:left-0 sm:right-0 sm:top-0 z-10 bg-base items-center px-4 py-2">
      <Link to="/">
        <h1 className="text-letra font-black text-4xl px-4 text-center">
          GamaDev
        </h1>
      </Link>
      <ListOfLinks />
    </nav>
  )
}

export default NavBar
