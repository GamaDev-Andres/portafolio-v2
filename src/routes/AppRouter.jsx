import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
