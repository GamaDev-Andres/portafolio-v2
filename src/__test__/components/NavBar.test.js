/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import NavBar from '../../components/NavBar'
import AppRouter from '../../routes/AppRouter'

describe('probando <NavBar/>', () => {
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter>

        <NavBar />
      </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h1').text().trim()).toBe('GamaDev')
    expect(wrapper.find('ListOfLinks').exists()).toBe(true)
  })
  test('no debe de mostrarse en ruta "project"', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/project/BARBERAPP']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider>
      </MemoryRouter>
    )
    expect(wrapper.find('NavBar').exists()).toBe(false)
  })
})
