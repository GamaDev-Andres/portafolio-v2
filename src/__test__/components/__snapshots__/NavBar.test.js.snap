// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <NavBar/> debe montarse correctamente 1`] = `
<nav
  className="sm:sticky w-full sm:top-0 sm:left-0 z-10"
>
  <div
    className="transition-colors duration-150 flex justify-center sm:justify-between relative sm:flex-col md:flex-row max-w-[1256px] mx-auto bg-base dark:bg-baseDark items-center px-4 py-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="text-letra dark:text-letraDark font-black text-4xl px-4 text-center"
      >
        GamaDev
      </h1>
    </a>
    <div
      className="flex items-center dark:text-letraDark  justify-end gap-4"
    >
      <div
        className="sm:gap-4 flex-grow sm:flex-grow-0 hidden justify-end sm:flex"
      >
        <a
          className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
          href="/#inicio"
        >
          <i
            className="sm:mr-2 fa-solid fa-house"
          />
          <span
            className="false"
          >
            inicio
          </span>
        </a>
        <a
          className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
          href="/#sobremi"
        >
          <i
            className="sm:mr-2 fa-solid fa-user"
          />
          <span
            className="false"
          >
            sobre m√≠
          </span>
        </a>
        <a
          className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
          href="/#habilidades"
        >
          <i
            className="sm:mr-2 fa-solid fa-laptop-code"
          />
          <span
            className="false"
          >
            habilidades
          </span>
        </a>
        <a
          className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
          href="/#proyectos"
        >
          <i
            className="sm:mr-2 fa-solid fa-briefcase"
          />
          <span
            className="false"
          >
            proyectos
          </span>
        </a>
      </div>
      <button
        className=" z-20 absolute top-2 bottom-2 right-4 sm:static "
        onClick={[Function]}
      >
        <i
          className="w-5 h-5 text-black fa-solid fa-moon"
        />
      </button>
    </div>
  </div>
</nav>
`;
